cachito/web/models.py

```diff
@@ -1,11 +1,69 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
+from copy import deepcopy
 
 from cachito.web import db
 
 
+request_pkg_manager_table = db.Table(
+    'request_pkg_manager',
+    db.Model.metadata,
+    db.Column('request_id', db.Integer, db.ForeignKey('request.id')),
+    db.Column('pkg_manager_id', db.Integer, db.ForeignKey('pkg_manager.id'))
+)
+
+
 class Request(db.Model):
     """A Cachito user request."""
+    __tablename__ = 'request'
     id = db.Column(db.Integer, primary_key=True)
+    git_repo = db.Column(db.String, nullable=False)
+    git_ref = db.Column(db.String, nullable=False)
+    pkg_managers = db.relationship('PackageManager', secondary=request_pkg_manager_table)
 
     def __repr__(self):
         return '<Request {0!r}>'.format(self.id)
+
+    def to_json(self):
+        pkg_managers = [pkg_manager.to_json() for pkg_manager in self.pkg_managers]
+        return {
+            'id': self.id,
+            'git_repo': self.git_repo,
+            'git_ref': self.git_ref,
+            'pkg_managers': pkg_managers,
+        }
+
+    @classmethod
+    def from_json(klass, kwargs):
+        # Validate all required parameters are present
+        required_params = {'git_repo', 'git_ref', 'pkg_managers'}
+        missing_params = required_params - set(kwargs.keys())
+        if missing_params:
+            raise ValueError('Missing required parameter(s): {}'.format(', '.join(missing_params)))
```

Comment:
It'd be nice to create a custom exception class, and add an error handler in Flask so that it automatically returns a JSON error to the user like Estuary does:
https://github.com/release-engineering/estuary-api/blob/master/estuary/app.py#L115

Reply:
Created shared `ValidationError` exception. I placed this directly under `cachito` module because it's generic enough to apply to either `web` or `worker`.